rdt/transformers/pii/anonymizer.py

```diff
@@ -0,0 +1,138 @@
+    def check_provider_function(provider_name, function_name):
+        """Check that the provider and the function exist.
+
+        Attempt to get the provider from ``faker.providers`` and then get the ``function``
+        from the provider object. If one of them fails, it will raise an ``AttributeError``.
+
+        Raises:
+            ``AttributeError`` if the provider or the function is not found.
+        """
+        module = getattr(faker.providers, provider_name)
+        if provider_name == 'BaseProvider':
+            getattr(module, function_name)
+
+        else:
+            provider = getattr(module, 'Provider')
+            getattr(provider, function_name)
+
+    def __init__(self, provider_name=None, function_name='lexify', function_kwargs=None,
+                 locales=None, missing_value_replacement=None, model_missing_values=False):
+        self.data_length = None
+        self.provider_name = provider_name if provider_name else 'BaseProvider'
+        self.function_name = function_name
+        self.function_kwargs = deepcopy(function_kwargs) if function_kwargs else {}
+        self.check_provider_function(self.provider_name, self.function_name)
+
+        self.missing_value_replacement = missing_value_replacement
+        self.model_missing_values = model_missing_values
+
+        self.locales = locales
+        self.faker = faker.Faker(locales)
+        self._function = getattr(self.faker, function_name)
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        output_types = {}
+        if self.null_transformer and self.null_transformer.models_missing_values():
+            output_types['is_null'] = 'float'
+
+        return self._add_prefix(output_types)
+
+    def _fit(self, columns_data):
+        self.null_transformer = NullTransformer(
+            self.missing_value_replacement,
+            self.model_missing_values
+        )
+        self.null_transformer.fit(columns_data)
+        self.data_length = len(columns_data)
+
+    def _transform(self, columns_data):
+        if self.null_transformer and self.null_transformer.models_missing_values():
+            return self.null_transformer.transform(columns_data)[:, 1].astype(float)
+
+        return None
+
+    def _reverse_transform(self, columns_data):
+        reverse_transformed = np.array([
+            self._function(**self.function_kwargs)
+            for _ in range(self.data_length)
+        ], dtype=object)
+
+        if self.null_transformer.models_missing_values():
+            reverse_transformed = np.column_stack((reverse_transformed, columns_data))
+
+        return self.null_transformer.reverse_transform(reverse_transformed)
+
+    def __repr__(self):
+        """Represent the class instance only with the modified default arguments."""
+        printable = [
```

Comment:
minor: is there a way to get this list programmatically from the `__init__` method? In case any parameters change

Reply:
Yes there it is 😅 , I have also changed the `BaseTransformer` one because of the following 'issue' that I found:

With the current implementation:
```python
class Foo:
    def __init__(self, param=None):
        self.param = param
        self.object = object()
        
    def __repr__(self):
        class_name = self.__class__.__name__
        args = []
        default = self.__class__()
        for arg, value in self.__dict__.items():
            if default.__dict__[arg] != value:
                if isinstance(value, str):
                    value = f"'{value}'"
                args.append(f'{arg}={value}')

        args_string = ', '.join(args)
        return f'{class_name}({args_string})'
```
When running this code:
```
Foo('a')
Foo(param='a', object=<object object at 0x7f3e6ceca620>)
```

However the `object` itself shouldn't be appearing there, it doesn't form part of the `__init__`. In order to fix it we can use `inspect` to check which are the `init` parameters and the defaults if there are any:
```python
def __repr__(self):
        class_name = self.__class__.__name__
        custom_args = []
        args = inspect.getfullargspec(self.__init__)
        keys = args.args[1:]
        defaults = args.defaults or []
        defaults = dict(zip(keys, defaults))
        instanced = {key: getattr(self, key) for key in keys}

        if defaults == instanced:
            return f'{class_name}()'

        for arg, value in instanced.items():
            if defaults[arg] != value:
                value = f"'{value}'" if isinstance(value, str) else value
                custom_args.append(f'{arg}={value}')

        args_string = ', '.join(custom_args)
        return f'{class_name}({args_string})'
```